enhancements/dns/externaldns-operator.md

```diff
@@ -0,0 +1,467 @@
+
+ExternalDNS could be configured to ignore all source resources with a static annotation determined by the operator,
+such as `externaldns.operator.io/ignore`. This would be useful if a user does not want to publish DNS for some specific resources.
+
+The ExternalDNS Operator CRD could also be expanded to include a field for selecting target nodes for ExternalDNS to run.
+This field could follow along with existing node placement APIs, such as the
+[IngressController's NodePlacement field](https://github.com/openshift/api/blob/master/operator/v1/types_ingress.go#L149).
+
+* Introduce a CRD for managing DNS records that satisfies the ExternalDNS
+base [CRD source](https://github.com/kubernetes-sigs/external-dns/blob/master/docs/contributing/crd-source.md).
+__Note:__ The [ingressoperator/dnsrecord](https://github.com/openshift/api/blob/master/operatoringress/v1/types.go) CRD
+could easily be expanded to meet the requirements of the ExternalDNS CRD source option.
+
+As an example, here is what a compatible DNSRecord Custom Resource instance may look like:
+
+```yaml
+apiVersion: operator.externaldns.k8s.io/v1alpha1
+kind: DNSRecord
+metadata:
+  name: default-wildcard
+  namespace: openshift-ingress-operator
+spec:
+  dnsName: 'foo.example.com'
+  providerSpecific:
+  - name: my-provider-setting-name
+    value: my-provider-setting-value
+  recordTTL: 30
+  recordType: A
+  targets:
+  - 1.2.3.4
+status:
+  observedGeneration: 1
+```
+
+This manifest creates a DNS A record for the hostname `foo.example.com` that resolves to address `1.2.3.4` in the configured
+provider, based on the resource's Domain. Additional `spec` and/or `status` fields may be introduced based on requirements.
+Note the optional `providerSpecific` field, in which name/value pairs can be added as necessary.
+
+Note that currently, the ExternalDNS CRD source implementation only updates the `observedGeneration` status subfield.
+An [issue has been opened upstream](https://github.com/kubernetes-sigs/external-dns/issues/2092) to better understand
+upstream's intentions with CRD source status fields. ExternalDNS does provide adequate debug logs, should a record fail to update,
+in addition to a few [high-level metrics](https://github.com/kubernetes-sigs/external-dns/blob/master/controller/controller.go).
+However, status fields to reflect the availability of the requested record in the given provider would be useful for other controllers running
+in a cluster, as well as for cluster administrators who want to verify that their requested resources have been created properly.
+
+Currently, the upstream ExternalDNS CRD source only supports record creation for A or CNAME type records. See
+[upstream issue 164](https://github.com/kubernetes-sigs/external-dns/issues/164) for more context to that end. Contributions may need to made to
+the upstream CRD source implementation to allow for other record types to be created via the DNSRecord CRD on OpenShift, such as `TXT` or `AAAA` records.
```

Comment:
With these known limitations, should thee EP focus on Route and Service sources? The EP can be updated or a separate EP created in the future to add CRD support.

Reply:
I think this EP should include CRD support, since improvements to the Upstream CRD source could be worked on as a part of this EP.